Firestore/Swift/Source/PropertyWrapper/FirestoreQuery.swift

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import FirebaseFirestore
+
+public enum DecodingFailureStrategy {
+  case ignore
+  case raise // make this the default behaviour
+}
+
+/// A property wrapper that listens to a Firestore collection.
+///
+/// Consider the following example:
+///
+///     struct ContentView: View {
+///       @FirestoreQuery(
+///         collectionPath: "fruits",
+///         predicates: [.whereField("isFavourite", isEqualTo: true)]
+///       ) var fruits: [Fruit]
+///
+///       var body: some View {
+///         List(fruits) { fruit in
+///           Text(fruit.name)
+///         }
+///       }
+///     }
```

Comment:
The code example in the doc comment is awesome. WDYT about including another one for the Result type implementation, or linking to your sample app?

Reply:
Good idea! Added.